lib/GridItem.jsx

```diff
@@ -596,22 +582,67 @@ export default class GridItem extends React.Component<Props, State> {
       size.width,
       size.height,
       x,
-      y
+      y,
+      handle
     );
 
     // minW should be at least 1 (TODO propTypes validation?)
     minW = Math.max(minW, 1);
 
-    // maxW should be at most (cols - x)
-    maxW = Math.min(maxW, cols - x);
-
     // Min/max capping
     w = clamp(w, minW, maxW);
     h = clamp(h, minH, maxH);
 
-    this.setState({ resizing: handlerName === "onResizeStop" ? null : size });
+    if (node) {
+      const currentLeft = position.left;
+      const currentTop = position.top;
+      const currentWidth = position.width;
+      const currentHeight = position.height;
+
+      const constrainWidth = () => {
+        if (currentLeft + size.width > this.props.containerWidth) {
+          size.width = currentWidth;
+        }
+      };
+
+      const constrainHeight = () => {
+        if (size.top <= 0) {
+          size.height = currentHeight;
+        }
+      };
+
+      const constrainLeft = () => {
+        if (size.left <= 0) {
+          size.left = currentLeft;
+          size.width = currentWidth;
+        }
+      };
+
+      if (["sw", "w", "nw", "n", "ne", "e", "se"].indexOf(handle) !== -1) {
```

Comment:
`"s"` deliberately excluded? 

Nested conditional is a bit ugly

Reply:
Yea delibrately excluded since there is not additional modifications that need to be made. I'll clean up this whole block of logic, this was stilla remnant of Ludovics work.